python/Ganga/GPIDev/Base/VPrinter.py

```diff
@@ -316,27 +320,28 @@ def summary_print(obj, out=None, interactive=False):
             print('[]', end=' ', file=out)
         else:
             outString = '['
-            count = 0
-            for x in _obj:
+            outStringList = []
+            for x in obj:
                 if isType(x, GangaObject):
-                    sio = StringIO()
+                    sio =StringIO()
                     stripProxy(x).printSummaryTree(0, 0, '', out=sio)
                     result = sio.getvalue()
                     # remove trailing whitespace and newlines
-                    outString += result.rstrip()
+                    outStringList.append(result.rstrip())
                 else:
-                    result = str(x)
                     # remove trailing whitespace and newlines
-                    outString += result.rstrip()
-                count += 1
-                if count != obj_len:
-                    outString += ', '
+                    outStringList.append(str(x).rstrip())
+                # Check if we're at the last element or not and add commas
+                if len(outStringList)+1 != 2*obj_len:
+                    outStringList.append(', ')
+            outString += ''.join(outStringList)
```

Comment:
I managed to get some speed improvement from this when testing but it did depend on doing something lightweight to determine if the additional ',' character needs to be added. Would have to go looking for the benchmarks but this helped when generating a very large list I found